src/PoolBuffer.cpp

```diff
@@ -32,42 +32,71 @@
 /**
  * \file PoolBuffer.cpp
  * \author Chris Chafe
- * \date May 2021
+ * \date May-Sep 2021
  */
 
 // EXPERIMENTAL for testing in JackTrip v1.4.0
-// runs ok from FPP 16 up to 512, but don't try 1024 yet
+// runs ok from FPP 16 up to 256, but don't try 512 or 1024 yet
 // in / out channels are the same -- mono, stereo and -n3 tested fine
 
-// for example, server
-// jacktrip -S --udprt  -p1 --bufstrategy 3 -q2
-// and client
-// jacktrip -C cmn9.stanford.edu --udprt --bufstrategy 3 -q2
+// jacktrip -S --udprt -p1 --bufstrategy 3 -q33
+// PIPEWIRE_LATENCY=32/48000 ./jacktrip -C cmn9.stanford.edu --udprt --bufstrategy 3 -q3
+
+// local loopback test with 4 terminals running and the following jmess file
+// jacktrip -S --udprt --nojackportsconnect -q1 --bufstrategy 3
+// jacktrip -C localhost --udprt --nojackportsconnect -q1  --bufstrategy 3
+// jack_iodelay
+// jmess -c delay.xml
+// ---------delay.xml----------
+//<jmess>
+//  <connection>
+//    <output>localhost:receive_1</output>
+//    <input>jack_delay:in</input>
+//  </connection>
+//  <connection>
+//    <output>__1:receive_1</output>
+//    <input>__1:send_1</input>
+//  </connection>
+//  <connection>
+//    <output>__1:receive_2</output>
+//    <input>__1:send_2</input>
+//  </connection>
+//  <connection>
+//    <output>jack_delay:out</output>
+//    <input>localhost:send_1</input>
+//  </connection>
+//</jmess>
+
+// tested loss and jitter impairments with
+// sudo tc qdisc add dev lo root netem loss 1%
+// sudo tc qdisc del dev lo root netem loss 1%
+// sudo tc qdisc add dev lo root netem slot distribution pareto 0.5ms 0.5ms
+// sudo tc qdisc del dev lo root netem slot distribution pareto 0.5ms 0.5ms
 
 #include "PoolBuffer.h"
 
+#include <QTextStream>
 #include <iomanip>
+#include <sstream>
 
 #include "jacktrip_globals.h"
 using std::cout;
 using std::endl;
 using std::setw;
 
-#define STDDEVINDOW     200  // packets
-#define STDDEV2POOLSIZE 30.0
-#define MAXPOOLSIZE     6000  // insanely large pool, 2 sec FPP16
-#define HIST            6     // at FPP32
-#define OUT(x, ch, s)   sampleToBits(x, ch, s)
+#define HIST          6  // at FPP32
+#define OUT(x, ch, s) sampleToBits(x, ch, s)
```

Comment:
HIST can be a constexpr member variable. OUT doesn't gain anything.

Reply:
thanks!